src/radical/saga/adaptors/lsf/lsfjob.py

```diff
@@ -71,46 +67,38 @@ def run(self):
 
         self.logger.info('start thread for %s', self.js.get_url())
         while not self._term.isSet():
-        
+
             try:
                 # do bulk updates here! we don't want to pull information
                 # job by job. that would be too inefficient!
-                jobs = self.js.jobs
-                for job in jobs:
+                for job in self.js._jobs:
+                    job_info = self.js._jobs[job]
 
                     # if the job hasn't been started, we can't update its
                     # state. we can tell if a job has been started if it
                     # has a job id
-                    job_info = self.js.jobs[job]
-                    if not job_info.get('job_id'):
-                        continue
-
-                    # we only need to monitor jobs that are not in a
-                    # terminal state, so we can skip the ones that are
-                    # either done, failed or canceled
-                    state = job_info['state']
-                    if state in rsj.FINAL:
-                        continue
-
-                    # job is not final and we got new info - replace job info
-                    new_info = self.js._job_get_info(job)
-                    self.js.jobs[job] = new_info
-
-                    # we only care to state updates though when the state
-                    # actually changed from last time
-                    new_state = new_info['state']
-                    if new_state == state:
-                        continue
-
-                    # fire job state callback if 'state' has changed
-                    self.logger.info("update Job %s (state: %s)"
-                                    % (job, new_state))
-                    job._api()._attributes_i_set('state', new_state,
-                                                 job._api()._UP, True)
+                    if job_info.get('id'):
+                        # we only need to monitor jobs that are not in a
+                        # terminal state, so we can skip the ones that are
+                        # either done, failed or canceled
+                        state = job_info['state']
+                        if state not in [api.DONE, api.FAILED, api.CANCELED]:
+
+                            new_info = self.js._job_get_info(job)
```

Comment:
`job_id` for _job_get_info

Reply:
`job` is the job ID.  I'll rename the variable to make this clear.